### PR TITLE
Panic when shred index exceeds the max per slot

### PR DESCRIPTION
**Pull request by sakridge**
#### Problem

Nodes keep making new shreds and broadcasting them after hitting the shred limit

#### Summary of Changes

Panic the node when this case happens

Fixes #

### CODE REVIEW

**Review by behzadnouri — APPROVED** [review]

**Review by behzadnouri — APPROVED** [review]

File: core/src/broadcast_stage/standard_broadcast_run.rs [hunk 1]
```diff
@@ -150,10 +156,15 @@ impl StandardBroadcastRun {
             Some(index) => index + 1,
             None => next_shred_index,
         };
+
+        // If the node is hitting this limit, it probably is forked from the network
+        // and creating a huge slot that will not be accepted by the network anyway.
```
Comment (t-nelson) [inline core/src/broadcast_stage/standard_broadcast_run.rs:161]:
```suggestion
```

maybe not the code to comment 😉

**Review by t-nelson — APPROVED** [review]
r+

**Review by t-nelson — DISMISSED** [review]

File: ledger/src/shred/shred_code.rs [hunk 2]
```diff
@@ -14,8 +14,7 @@ use {
 };
 
 const_assert_eq!(MAX_CODE_SHREDS_PER_SLOT, 32_768 * 17);
-pub(crate) const MAX_CODE_SHREDS_PER_SLOT: usize =
-    MAX_DATA_SHREDS_PER_SLOT * (ERASURE_BATCH_SIZE[1] - 1);
+pub const MAX_CODE_SHREDS_PER_SLOT: usize = MAX_DATA_SHREDS_PER_SLOT * (ERASURE_BATCH_SIZE[1] - 1);
```
Comment (behzadnouri) [inline ledger/src/shred/shred_code.rs:17]: Why is this modified?
Also I am reducing this in https://github.com/solana-labs/solana/pull/30543

**Review by behzadnouri — APPROVED** [review]

File: ledger/src/shred/shred_code.rs (hunk 2 above)
Comment (sakridge) [reply ledger/src/shred/shred_code.rs:17]: I just made it pub so I can import it into the other crate to check for excess coding shreds.

Comment (sakridge) [reply ledger/src/shred/shred_code.rs:17]: Value is not changed.

Comment (t-nelson) [reply ledger/src/shred/shred_code.rs:17]: sounds like the test (of the constant values) is in the wrong place, then?

Comment (sakridge) [reply ledger/src/shred/shred_code.rs:17]: sorry, are you saying move the MAX_CODE_SHREDS_PER_SLOT into the core crate?

Comment (t-nelson) [reply ledger/src/shred/shred_code.rs:17]: nah. that the test in core doesn't really care what the values are and that if a more specific test around the constants is warranted, it should be in the ledger crate

Comment (sakridge) [reply ledger/src/shred/shred_code.rs:17]: Yes, the test code doesn't use the constant at all, just in the validator code when I call that function I use the real constant value.

Comment (t-nelson) [reply ledger/src/shred/shred_code.rs:17]: ah... derp. sorry misread as in test there
Screenshot from 2023-03-03 16-51-20
